lib/vector-core/internal-event/src/lib.rs

```diff
@@ -0,0 +1,9 @@
+pub trait InternalEvent {
```

Comment:
Now that you're the proud owner of a crate would you be so kind as to add a documentation and clippy preamble? Here's an example from vector-core:

https://github.com/vectordotdev/vector/blob/c6c1630fd10cc05453e682615dd2d6643eabe6c7/lib/vector-core/src/lib.rs#L1-L22

The `allow` attributes you can drop here, I'm sure. Mostly we want to have good hygiene around documentation and lints and moving stuff around in the codebase is an excellent opportunity.

Reply:
Sure thing, thanks for the heads up.

Reply:
Much obliged. When that commit lands I'll be +1 here.

Reply:
Hah! Well, I guess my jokey comment means I own the `buffer` crate, so, uh.

Reply:
_yes_